### PR TITLE
Added class for hanlding logging instance and add logging to 'Log' class, system calls and lock

### LINKED ISSUE #178: Create logging to log command runs/output and debug
The daemon should log to /var/log/...
This should contain debug, if enabled, as well as failed command runs (or all command runs, if using debug)
This would allow for users to see what the daemon is doing.

### PR DESCRIPTION
**Pull request by MatthewJohn**
Fixes #178

### CODE REVIEW

**Comment by itdevreviewboard** [comment]
Build triggered. sha1 is merged.

**Comment by itdevreviewboard** [comment]
Build started sha1 is merged.

**Comment by itdevreviewboard** [comment]
Build finished.

File: source/usr/lib/mcvirt/logger.py [hunk 1]
```diff
@@ -123,22 +130,38 @@ def description(self):
...
+            str(self.finish_time), self.user or '', self.object_type or '', self.object_name or '',
+            self.method_name or ''
+        ]))
 
     def finish_error_unknown(self, exception):
         self.finish_time = datetime.now()
         self.status = LogState.FAILED
         self.exception_message = str(exception)
         self.exception_mcvirt = False
+        Syslogger.logger().debug('Command failed (Unknown Exception): %s' % ', '.join([
```
Comment (joe-singleton-itdev) [inline source/usr/lib/mcvirt/logger.py]: Should be error instead of debug

File: source/usr/lib/mcvirt/logger.py [hunk 2]
```diff
@@ -123,22 +130,38 @@ def description(self):
...
+            str(self.finish_time), self.user or '', self.object_type or '', self.object_name or '',
+            self.method_name or '', self.exception_message or ''
+        ]))
 
     def finish_error(self, exception):
         self.finish_time = datetime.now()
         self.status = LogState.FAILED
         self.exception_message = str(exception)
         self.exception_mcvirt = True
+        Syslogger.logger().debug(' Command failed (MCVirt Exception): %s' % ', '.join([
```
Comment (joe-singleton-itdev) [inline source/usr/lib/mcvirt/logger.py]: Should be error instead of debug

**Comment by itdevreviewboard** [comment]
Build triggered. sha1 is original commit.

**Comment by itdevreviewboard** [comment]
Build started sha1 is original commit.

**Comment by itdevreviewboard** [comment]
Build finished.

**Pull request merged**